src/deploy/functions/prepare.ts

```diff
@@ -1,22 +1,21 @@
 import * as clc from "cli-color";
 
-import { Options } from "../../options";
-import { ensureCloudBuildEnabled } from "./ensureCloudBuildEnabled";
-import { functionMatchesAnyGroup, getFilterGroups } from "./functionsDeployHelper";
-import { logBullet } from "../../utils";
-import { getFunctionsConfig, prepareFunctionsUpload } from "./prepareFunctionsUpload";
-import { promptForFailurePolicies, promptForMinInstances } from "./prompts";
 import * as args from "./args";
 import * as backend from "./backend";
 import * as ensureApiEnabled from "../../ensureApiEnabled";
 import * as functionsConfig from "../../functionsConfig";
 import * as functionsEnv from "../../functions/env";
+import * as runtimes from "./runtimes";
+import { Options } from "../../options";
+import { ensureCloudBuildEnabled, ensureSecretAccess, maybeEnableAR } from "./ensure";
+import { functionIdsAreValid, secretsAreValid } from "./validate";
+import { functionMatchesAnyGroup, getFilterGroups } from "./functionsDeployHelper";
+import { logBullet } from "../../utils";
+import { getFunctionsConfig, prepareFunctionsUpload } from "./prepareFunctionsUpload";
+import { promptForFailurePolicies, promptForMinInstances } from "./prompts";
 import { previews } from "../../previews";
 import { needProjectId } from "../../projectUtils";
 import { track } from "../../track";
-import * as runtimes from "./runtimes";
```

Comment:
but.. but... go-style `module.function` made it easier to keep track of where all this damned code came from =/

Ah well. Guess it's time for me to be more hands off as a manager.

Reply:
Lol funny story - the reason I do this, here and onward, is because IntelliJ defaults doing per-object imports 🤷🏼 . I see your point and will revert the changes.